client/asset/eth/eth_test.go

```diff
@@ -1262,6 +1274,87 @@ func TestMaxOrder(t *testing.T) {
 	}
 }
 
+func TestRefund(t *testing.T) {
+	node := &testNode{}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	eth := &ExchangeWallet{
+		node: node,
+		ctx:  ctx,
+	}
+	tests := []struct {
+		name            string
+		contract        dex.Bytes
+		maxFeeRate      uint64
+		isRefundable    bool
+		isRefundableErr error
+		refundErr       error
+		wantErr         bool
+	}{
+		{
+			name:         "ok",
+			contract:     versionedBytes(0, encode.RandomBytes(32)),
+			isRefundable: true,
+			maxFeeRate:   100,
+		},
+		{
+			name:            "is refundable error",
+			contract:        versionedBytes(0, encode.RandomBytes(32)),
+			isRefundable:    true,
+			isRefundableErr: errors.New(""),
+			maxFeeRate:      100,
+			wantErr:         true,
+		},
+		{
+			name:         "is refundable false",
+			contract:     versionedBytes(0, encode.RandomBytes(32)),
+			isRefundable: false,
+			maxFeeRate:   100,
+			wantErr:      true,
+		},
+		{
+			name:         "refund error",
+			contract:     versionedBytes(0, encode.RandomBytes(32)),
+			isRefundable: true,
+			refundErr:    errors.New(""),
+			maxFeeRate:   100,
+			wantErr:      true,
+		},
+		{
+			name:         "secret hash wrong length",
+			contract:     versionedBytes(0, encode.RandomBytes(31)),
+			isRefundable: true,
+			maxFeeRate:   100,
+			wantErr:      true,
+		},
+		{
+			name:         "cannot parse versioned bytes",
+			contract:     []byte{},
+			isRefundable: true,
+			maxFeeRate:   100,
+			wantErr:      true,
+		},
+	}
+
+	for _, test := range tests {
+		node.refundable = test.isRefundable
+		node.isRefundableErr = test.isRefundableErr
+		node.refundErr = test.refundErr
+
+		_, err := eth.Refund(nil, test.contract, test.maxFeeRate)
+		if test.wantErr {
+			if err == nil {
+				t.Fatalf(`%v: expected error but did not get: %v`, test.name, err)
+			}
+			continue
+		}
+		if err != nil {
+			t.Fatalf(`%v: unexpected error: %v`, test.name, err)
+		}
```

Comment:
@buck54321 I would want to check that the hash of the tx used to do the refund is the same as the coin id returned from `Refund`. I'm wondering what you think about what we were discussing in this PR:
https://github.com/decred/dcrdex/pull/1318

Personally I would just add a `lastRefundTxHash` to the `testNode` and use that to compare, but if you're planning to do something else, I'll hold off on this.

Reply:
Please do @martonp. I got the things I though were most important in #1318, but if you see a place to improve, you should.